statsmodels/stats/contingency_tables.py

```diff
@@ -1402,3 +1484,1238 @@ def cochrans_q(x, return_object=True):
+        NOT_AVAILABLE = "Not Available From Rao-Scott Method"
+        if rows_are_multiple_response and columns_are_multiple_response:
+            spmi_test = self._test_for_single_pairwise_mutual_independence_using_rao_scott_2
+            table_p_value = spmi_test(self.row_factors[0], self.column_factors[0])
+            result = self._build_MRCV_result(table_p_value, NOT_AVAILABLE, "Rao-Scott", "SPMI")
+            return result
+        elif columns_are_multiple_response:
+            table_p_value = mmi_test(self.row_factors[0], self.column_factors[0])
+            result = self._build_MRCV_result(table_p_value, NOT_AVAILABLE, "Rao-Scott", "MMI")
+            return result
+        elif rows_are_multiple_response:
+            table_p_value = mmi_test(self.column_factors[0], self.row_factors[0])
+            result = self._build_MRCV_result(table_p_value, NOT_AVAILABLE, "Rao-Scott", "MMI")
+            return result
+        else:
+            single_response_table = Table(self.table)
+            return single_response_table.test_nominal_association()
+
+    @classmethod
+    def _extract_and_validate_factors(cls, column_factors, row_factors):
+        """
+        Make sure that the factors that the user passed into the initial MultipleResponseContingencyTable
+        initializer are valid and that the combination of single and multiple
+        response variables is currently supported. Also, since we don't currently
+        support multiple factors on a single axis we enforce that only one
+        has been passed.
+
+        Also de-duplicate factor level names and pivot factors to wide
+        orientation (if necessary)
+
+        Parameters
+        ----------
+        row_factors : list of Factor instances
+            Factors to use on the rows
+        column_factors : list of Factor instances
+            Factors to use in the columns
+        Return
+        ------
+        (Factor, Factor)
+            A single factor instance to use on the columns and a
+            single factor instance to use on the rows.
+        """
+        try:
+            try:
+                if len(row_factors) > 1:
+                    msg = "we don't currently support tables with more than one factor on the rows"
+                    raise NotImplementedError(msg)
+                row_factor = row_factors[0]
+            except TypeError:
+                if isinstance(row_factors, Factor):
+                    row_factor = row_factors
+                else:
+                    msg = "row_factors must be either a list of Factors or a Factor instance"
+                    raise NotImplementedError(msg)
+            try:
+                if len(column_factors) > 2:
+                    msg = "we don't currently support tables with more than one factor on the columns"
+                    raise NotImplementedError(msg)
+                column_factor = column_factors[0]
+            except TypeError:
```

Comment:
just a general answer:

We don't have a strict or clearly defined policy for input verification. Traditionally we have avoided expensive checks in favor of performance and relying on users to do some checks.
However, we do a lot of type conversion in the initial parts of a function or in the data handling of the models. Those are partially cheap and partially a trade-off so we can allow different types of inputs. 
If I remember correctly what I was thinking, then this type of input ambiguity or flexibility should be resolved in the `__init__` instead of individual methods, so that the methods can really on clearly defined attributes.
For example an isinstance check or check for specific attributes is still cheap relative to the computations that we have in the functions and classes. But in several places we also have keywords for the user to choose the treatment (e.g. `missing`) or to provide information that avoids the internal check (`has_const` ?).